### ISSUE TITLE
New features ideas

### ISSUE THREAD

**Issue by jakubvojacek** [andipaetzold/tttt-discord#814]
Hello @andipaetzold 

I was thinking of possible improvements that myself and my team could use. While I could implement some of them and send a PR, I'd like to discuss them first and maybe find a better approach.

We had problems with people pressing skip button because they were dead and wanted to rest once. But the problem was that they pressed the button when it wasnt their turn so they just skipped the current athlete, not themselves. You can only skip yourself if you wait for your turn and then press skip. But that kind of breaks the rotation as for a short period time, there'd be nobody pulling and the next person would just get cue `<Go name>` instead of the countdown.

There are multiple ways to tackle this

- the reaction button will only work for the person pressing it, pressing skip at any time will result in the person who pressed it being skipped one time. Pressing plus will add time for his next pull
- add DS (perhaps /timer ds <athlete>), and for him, the reaction button would work as they work now ?

Or maybe even better add multiple status messages, one per each athlete and each of the status messages would have the reaction button and those would control the given athlete only? It would then work like the web UI https://andipaetzold.github.io/tttt/ where its really easy to toast someone even when its not his turn. Last tuesday I had two riders who wanted to get toasted and since it wasnt their turn, I had to switch from zwift to discord and type /timer toast <name> which isnt very easy when pushing 350 watts :) clicking on icon would have been faster

<img width="865" alt="Screenshot 2023-11-30 at 8 52 20" src="https://github.com/andipaetzold/tttt-discord/assets/3463456/b41a9535-12a7-40c6-b861-6912a17ad85e">

a quick mockup above - we could even add minus button, make it denser so that more athletes would be visible at one time etc

**Comment by andipaetzold** [comment]
Thanks for the ideas!

# Allow athletes to skip themselves

When you configure the athletes with `/timer athletes`, you can @-mention discord users. The bot will then announce the user name. When that user presses the skull "button" the athlete gets toasted and skipped until the button is pressed again. This can be done at any time and the athlete doesn't have to wait for their round. When pressing the skull and they are already in front, their turn ends immediately.

Are you aware of that behavior? And wouldn't that solve your issue?

# DS view

I like the idea of a different view for the DS. It must be a separate command to not flood everybody with status messages. Ideally, the DS would call that command in a separate channel, so the competing athletes only see a single status message and don't get overloaded with buttons and changing texts.

**Comment by jakubvojacek** [comment]
> Are you aware of that behavior? And wouldn't that solve your issue?

yes aware of this. But most people dont use PC to ride on, they use ATV, apple tv etc and discord they use on phone with headponse for example. So either they're unable to press it all or they press it repeatedly etc.

I am usually the DS writing commands and it'd be much easier to click on skull than to type `/timer toast @xxxxxxx`

Anyhow, last TTT, I only had to toast two people and didnt have to change anything else (pull duration etc). We only used this bot twice so far in race scenarios - not sure how much and how often I will need to adjust. It would be quite a big development with potentially not so much use. I will first test out few races and get back.
